tests/infrastructure/tekton/test_tekton_pipeline_disk_uploader.py

```diff
@@ -0,0 +1,166 @@
+
+
+@pytest.fixture()
+def created_fedora_dv(admin_client, golden_images_namespace, custom_pipeline_namespace):
+    with create_dv(
+        dv_name="fedora-dv-disk-uploader",
+        namespace=custom_pipeline_namespace.name,
+        source=PVC,
+        source_pvc=DataSource(
+            name=OS_FLAVOR_FEDORA, namespace=golden_images_namespace.name
+        ).instance.spec.source.pvc.name,
+        source_namespace=golden_images_namespace.name,
+        size="35Gi",
+        client=admin_client,
+        storage_class=py_config["default_storage_class"],
+    ) as dv:
+        dv.wait_for_dv_success()
+        yield dv
+
+
+@pytest.fixture()
+def vm_for_disk_uploader(custom_pipeline_namespace, admin_client, created_fedora_dv):
+    with VirtualMachineForTests(
+        name="fedora-vm-diskuploader",
+        namespace=custom_pipeline_namespace.name,
+        client=admin_client,
+        data_volume=created_fedora_dv,
+        vm_instance_type=VirtualMachineClusterInstancetype(name="u1.small"),
+        vm_preference=VirtualMachineClusterPreference(name=OS_FLAVOR_FEDORA),
+        run_strategy=VirtualMachine.RunStrategy.ALWAYS,
+    ) as vm:
+        wait_for_running_vm(vm=vm, wait_for_interfaces=False, check_ssh_connectivity=False)
+        yield vm
+
+
+@pytest.fixture()
+def stopped_vm_for_disk_uploader(vm_for_disk_uploader):
+    vm_for_disk_uploader.stop(wait=True)
+    yield
+
+
+@pytest.fixture()
+def pipeline_disk_uploader(
+    admin_client,
+    custom_pipeline_namespace,
+):
+    with Pipeline(
+        name="pipeline-disk-uploader",
+        namespace=custom_pipeline_namespace.name,
+        client=admin_client,
+        tasks=pipeline_tasks,
+        params=pipeline_params,
+    ) as pipeline:
+        if pipeline.exists:
+            yield pipeline
+        raise ResourceNotFoundError(f"Pipeline {pipeline.name} not found")
+
+
+@pytest.fixture()
+def pipelinerun_for_disk_uploader(
+    admin_client,
+    custom_pipeline_namespace,
+    quay_disk_uploader_secret,
+    pipeline_disk_uploader,
+    created_fedora_dv,
+    vm_for_disk_uploader,
+):
+    pipeline_run_params = {
+        "EXPORT_SOURCE_KIND": "vm",
+        "EXPORT_SOURCE_NAME": vm_for_disk_uploader.name,
+        "VOLUME_NAME": created_fedora_dv.name,
+        "IMAGE_DESTINATION": "quay.io/openshift-cnv/tekton-tasks",
+        "PUSH_TIMEOUT": TIMEOUT_5MIN,
+        "SECRET_NAME": quay_disk_uploader_secret.name,
+    }
+
+    with PipelineRun(
+        name="pipelinerun-disk-uploader",
+        namespace=custom_pipeline_namespace.name,
+        client=admin_client,
+        params=pipeline_run_params,
+        pipelineref=pipeline_disk_uploader.name,
+    ) as pipelinerun:
+        pipelinerun.wait_for_conditions()
+        yield pipelinerun
+
+
+@pytest.mark.usefixtures("extracted_kubevirt_tekton_resources", "processed_yaml_files")
+@pytest.mark.polarion("CNV-11721")
+def test_disk_uploader_pipelinerun(
+    vm_for_disk_uploader,
+    stopped_vm_for_disk_uploader,
```

Comment:
do you really need to create VM, wait when it is running and after that stop the vm?
may be create VM in non-running state would be enough?

Reply:
I want to try a use case where VM is running and pvc in use . https://issues.redhat.com/browse/CNV-54663
Currently we just keep waiting till pipeline timeout. When we have this fix , I should be able to cover that as well in automation. That's the reason i want a running VM.